src/gluonts/dataset/common.py

```diff
@@ -129,16 +129,18 @@ def infer_file_type(path):
     return None
 
 
-def _glob(path: Path, pattern="*", levels=1):
+def _glob(path: Path, pattern="*", levels=None):
```

Comment:
Should we limit the number of `levels` by default?

If misused, this can be become a performance issues. Maybe we set this to a lower number and set it to `None` in `shell`?